src/home/homework.c

```diff
@@ -1901,6 +1913,41 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 return log_error_errno(r, "[%s:%u:%u] Failed to parse JSON data: %m", json_path, line, column);
 
+        fdmap = json_variant_by_key(v, HOMEWORK_BLOB_FDMAP_FIELD);
+        if (fdmap) {
+                r = hashmap_ensure_allocated(&blobs, &blob_fd_hash_ops);
+                if (r < 0)
+                        return log_oom();
+
+                JSON_VARIANT_OBJECT_FOREACH(blob_filename, blob_fd_variant, fdmap) {
+                        _cleanup_free_ char *filename = NULL;
+                        _cleanup_close_ int fd = -EBADF;
+
+                        assert(json_variant_is_integer(blob_fd_variant));
+                        assert(json_variant_integer(blob_fd_variant) >= 0);
+                        assert(json_variant_integer(blob_fd_variant) <= INT_MAX);
```

Comment:
feels weird to me to make json type checks via asserts.

Reply:
On one hand yes, but on the other hand it's private API glue between homed and homework and it really is a programmer error somewhere if the type check fails

Reply:
(in other words, if these type checks fail something's really gone wrong and crashing is probably more correct than logging a nice error message and exiting cleanly, IMO)

Reply:
To ensure that we never accidentally pass the field into homework from the user, I added code to make homed outright reject any user record that has the internal field set